docs/user-documentation/file_viewers.md

```diff
@@ -0,0 +1,84 @@
+# File Viewers
+
+_last updated on 03-12-2020_
+
+
+## What are viewers?
+
+[Viewers](../glossary#viewer) allow site builders to display files in interactive javascript-based widgets, that provide functionality like zooming in/out, turning pages, playing/pausing, viewing in full screen, etc. 
+
+In Drupal, a common way to implement a viewer is through a [module](./glossary/#module) that provides the javascript library, and a field formatter that uses that library. Usually the field formatter can be used on File fields.  
+
+Viewers that are known to work with Islandora 8 include:
+
+* [OpenSeadragon](https://openseadragon.github.io/), via the Drupal module [OpenSeadragon](https://github.com/Islandora/openseadragon) (maintained by the Islandora Foundation)
+* [pdf.js](https://github.com/mozilla/pdf.js), via the Drupal contrib module [PDF](https://www.drupal.org/project/pdf)
+
+Both are included with Islandora Defaults.
+
+## How are viewers configured?
+
+In the generic Islandora 8 content modelling scenario, we often want the viewer to appear on a _"resource node"_ when the file itself is in a field on a Media that references that node. These layers of indirection make displaying viewers on nodes rather complex.
+
+## Example - Open Seadragon (in Islandora Defaults)
+
+Here is how viewers are implemented in Islandora Defaults. This example uses OpenSeadragon, but pdf.js is configured with the same method. This method was chosen because it uses pure configuration and no extra code. It relies on users tagging nodes with appropriate "display hint" terms to trigger Contexts to switch the node's view mode to one that includes an EVA field, which renders the appropriate Media's file in the viewer.
+
+### Expected behaviour for this scenario:
+
+Create a Repository Item, and tag it with "Open Seadragon" (under Display Hints). Add a Media of type "Image" to that node, and tag it with "Original File" (under Media Use). Upload an image file that can be viewed in OpenSeadragon, for example, a jpeg-2000 file, and save (publish) the Media. Go to the Repository Item's page, and you should see your file displayed in the viewer.
```

Comment:
"...that can be processed by the Cantaloupe Image Server, for example, a Jpeg-2000, Tiff, or Jpeg file,..." 
OpenSeadragon is actually issuing IIIF calls to Cantaloupe, so what can be displayed will be determined by your Cantaloupe image processors.

Reply:
1 - can't upload a Tiff as an Image. Actually can't upload jp2 either (haha nobody caught that). (which is ridiculous)
2 - what's the relationship between the two? OpenSeadragon needs an image server, image servers can be configured to handle different file types? Is that important?

Reply:
Oh, I read right past the "Media of type 'Image'".... Both File and Image media types have the OpenSeadragon view mode because either can be used with the viewer (what matters is the actual file, not the Media type). Keep in mind that, to Drupal, Image media are those that can be rendered natively by a browser which excludes both Tiff and Jpeg-2000, which is why we put OpenSeadragon on both Image and File types. (Nearly all my Original Files are Tiffs, so I don't even think about Image for Original Files; not that this process cares....)

Cantaloupe is simply retrieving the bits it gets from the URL the OpenSeadragon viewer asks for and then does it's thing (it will do file identification and then select the processor you've configured it to work with). Technically, Cantaloupe can be configured with processors for PDFs as well as video files (although I haven't tested any of those).

Reply:
"to Drupal", images are "locally stored images" so we're not exactly using it as intended anyway.  I know why the decision was made, but I'm bitter about it because it makes our repository software especially unfriendly, in a way that causes people who actually have to use it from the front end get frustrated and feel stupid, when they have what are likely the most common file types that are dealt with in a digital repository. Not a conversation for this PR. I wanted to use a non-degenerate (in the math sense) example so I changed it all to use "File" media type instead of "Image".

Reply:
Piping in here to reference #1541 and how it might help with some UX around files.

Reply:
Yeah, as far as Drupal is concerned, anything accessed via Flysystem *is* local; but then, many system stretch the definition of "local". (Thanks, ☁️🖥!)

You are right, we devs can try to be clever and make things *super flexible*, but in the end make everything so fuzzy that users find themselves in a confused haze. 🌫☹️